.github/workflows/ci.yaml

```diff
@@ -0,0 +1,41 @@
+name: catkin_pkg-ci
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: ['*']
+
+jobs:
+    build:
+      strategy:
+        matrix:
```

Comment:
@cottsay With apologies that I'm experimenting rather than being consistent still I'd love your feedback on this approach to the build matrix. It's the smallest I could make it while still having any coherence. If you think this is more readable and maintainable than the big list of exclusions I'll open PRs to normalize it everywhere.